memory-viz/src/automate.ts

```diff
@@ -334,23 +348,28 @@ function compareByHeight(a, b) {
  * Returns a negative integer if 'a.id' is larger than 'b.id' (so, by definition of how sort uses the comparison
  * function, it will prioritize 'a' over 'b'), 0 if 'a' and 'b' have the same id's (WHICH SHOULD NOT HAPPEN),
  * and positive if 'b.id' is larger.
- * @param a - an object
- * @param b - another object
+ * @param {DrawnEntity} a - an object
+ * @param {DrawnEntity} b - another object
  * @returns {number} negative if 'a.id' is larger, 0 if a.id == b.id, and positive if 'b.id' is larger.
  */
-function compareByID(a, b) {
-    return a.id - b.id;
+function compareByID(a: DrawnEntity, b: DrawnEntity): number {
+    if (typeof a.id === "number" && typeof b.id === "number") {
+        return a.id - b.id;
+    }
+    if (typeof a.id === "string" && typeof b.id === "string") {
+        return a.id.localeCompare(b.id);
+    }
```

Comment:
Verify this behavior. The `id` can be both number and string

Reply:
@david-yz-liu Should I remove the `string` type from `id`? I haven't seen a single case where we use a string id.

Reply:
Yes it's fine to remove the `string` type from `id`